Assets/js/script.js

```diff
@@ -0,0 +1,282 @@
+// GLOBAL VARIABLES
+const lowerCase = [
+  "a",
+  "b",
+  "c",
+  "d",
+  "e",
+  "f",
+  "g",
+  "h",
+  "i",
+  "j",
+  "k",
+  "l",
+  "m",
+  "n",
+  "o",
+  "p",
+  "q",
+  "r",
+  "s",
+  "t",
+  "u",
+  "v",
+  "w",
+  "x",
+  "y",
+  "z",
+];
+
+const upperCase = [
+  "A",
+  "B",
+  "C",
+  "D",
+  "E",
+  "F",
+  "G",
+  "H",
+  "I",
+  "J",
+  "K",
+  "L",
+  "M",
+  "N",
+  "O",
+  "P",
+  "Q",
+  "R",
+  "S",
+  "T",
+  "U",
+  "V",
+  "W",
+  "X",
+  "Y",
+  "Z",
+];
+
+const numeric = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+
+const special = [
+  "!",
+  '"',
+  "#",
+  "$",
+  "%",
+  "&",
+  "'",
+  "(",
+  ")",
+  "*",
+  "+",
+  ",",
+  "-",
+  ".",
+  "/",
+  ":",
+  ";",
+  "<",
+  "=",
+  ">",
+  "?",
+  "@",
+  "[",
+  "\\",
+  "]",
+  "^",
+  "_",
+  "`",
+  "{",
+  "|",
+  "}",
+  "~",
+];
+
+// FUNCTION DECLARATIONS
+var generateBtn = document.querySelector("#generate");
+
+// check password length is a number (not a string)
+const validateIsNumber = function (passwordLength) {
+  const isNumber = typeof passwordLength;
+  return isNumber;
+};
+
+// check password length is >=8 and <= 128
+const validateNumberValue = function (passwordLength) {
+  const validValue = passwordLength >= 8 && passwordLength <= 128;
+  return validValue;
+};
+
+const getPasswordLength = function () {
+  const passwordLength = prompt(
+    "Please select a password length between 8 and 128 characters."
+  );
+  return Number(passwordLength);
+};
+
+// check if lowercase characters is chosen
+const getIsLowercase = function () {
+  const isLowercase = confirm("Include lowercase characters?");
+  return isLowercase;
+};
+
+// check if uppercase characters is chosen
+const getIsUppercase = function () {
+  const isUppercase = confirm("Include uppercase characters?");
+  return isUppercase;
+};
+
+// check if numeric characters is chosen
+const getIsNumeric = function () {
+  const isNumeric = confirm("Include numeric characters?");
+  return isNumeric;
+};
+
+// check if special characters is chosen
+const getIsSpecial = function () {
+  const isSpecial = confirm("Include special characters?");
+  return isSpecial;
+};
+
+// compile password criteria selections
+const getCriteria = function () {
+  // get password length
+  const passwordLength = getPasswordLength();
+  const isValid =
+    validateIsNumber(passwordLength) === "number" &&
+    validateNumberValue(passwordLength) === true;
```

Comment:
just like the below comment if its truthy you are checking you do no need to necessarily state it using the ==  you could write it this way `validateIsNumber(passwordLength) === "number" &&
    validateNumberValue(passwordLength)` instead but either way it will work but the later reduces the lines and makes it more cleaner